File: C_Stamina_and_Tasks.py
for _ in range(int(input())):
    n = int(input())
    c = []
    p = []
    for i in range(n):
        temp = input().split()
        c.append(int(temp[0]))
        p.append(int(temp[1])/100)
    
    total = c[-1]
    for i in range(n-2,-1,-1):
        if (1 - p[i]) * total + c[i] > total:
            total = (1 - p[i]) * total + c[i]
    
    print(total)

File: C_Stamina_and_Tasks.cpp
#include <bits/stdc++.h>


using namespace std;
double round1e7(double x) {
    return std::round(x * 1e7) / 1e7;
}

void solve()
{
    int n, temp;
    cin >> n;
    vector<long double> c;
    vector<long double> p;
    long double total, temp2;
    for (int i = 0; i < n; i++)
    {
        cin >> temp;
        c.push_back(temp);
        cin >> temp2;
        p.push_back(temp2 / 100);
    }
    total = c[n - 1];

    for (int i = n - 2; i >= 0; i--)
    {

        if ((1 - p[i]) * total + c[i] > total)
        {
            total = round1e7((1 - p[i]) * total + c[i]);
        }
    }

    cout << total << endl;
}

int main()
{
    int t = 1;
    cin >> t;
    while (t--)
    {
        solve();
    }

    return 0;
}
